internal/api/types.go

```diff
@@ -3,13 +3,22 @@ package api
 import (
 	"context"
 	"regexp"
+	"strings"
 
 	"github.com/replit/upm/internal/util"
 )
 
 // PkgName represents the name of a package, e.g. "flask" for Python.
 type PkgName string
```

Comment:
Why declare this type instead of simply using string?

Reply:
Opaque types are a good way to communicate intent through types as well as avoid bugs during refactors.

As an example, `map[PkgName]PkgInfo` means that all the symantics we want in a `PkgName` have been satisfied, presumably normalization being the most important.

That would keep `BeautifulSoup4` and `beautifulsoup4` from being duplicates in that map.

Reply:
👍